### ISSUE TITLE
Can't fully-finetuning with FSDP 

### ISSUE THREAD

**Issue by hahmad2008** [axolotl-ai-cloud/axolotl#1114]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

I fully finetuning using FSDP on TinyLama on three workers each with two GPUs. 
The problem that this way need 90 hours to complete the training. however when I use deepspeed stag 2 and stage 3 (zeros2 and zer3), the full-finetune took only three hours.

configuration.yaml:

```
base_model: TinyLlama/TinyLlama-1.1B-step-50K-105b
base_model_config: TinyLlama/TinyLlama-1.1B-step-50K-105b
...
fsdp:
  - full_shard
  - auto_wrap
fsdp_config:
  fsdp_offload_params: true
  fsdp_state_dict_type: FULL_STATE_DICT
  fsdp_transformer_layer_cls_to_wrap: LlamaDecoderLayer
```

Is there any problems here? why does FSDP take 90 hours? I kill the fine-tuning process.
FSDP logs training:
`0%|▎                               | 1/712 [10:58<90:00:48, 447.23s/it]`

### Current behaviour

I fully finetuning using FSDP on TinyLama on three workers each with two GPUs. 
The problem that this way need 90 hours to complete the training. however when I use deepspeed stag 2 and stage 3 (zeros2 and zer3), the full-finetune took only three hours.

configuration.yaml:

```
base_model: TinyLlama/TinyLlama-1.1B-step-50K-105b
base_model_config: TinyLlama/TinyLlama-1.1B-step-50K-105b
...
fsdp:
  - full_shard
  - auto_wrap
fsdp_config:
  fsdp_offload_params: true
  fsdp_state_dict_type: FULL_STATE_DICT
  fsdp_transformer_layer_cls_to_wrap: LlamaDecoderLayer
```

Is there any problems here? why does FSDP take 90 hours? I kill the fine-tuning process.
FSDP logs training:
`0%|▎                               | 1/712 [10:58<90:00:48, 447.23s/it]`

### Steps to reproduce

I fully finetuning using FSDP on TinyLama on three workers each with two GPUs. 
The problem that this way need 90 hours to complete the training. however when I use deepspeed stag 2 and stage 3 (zeros2 and zer3), the full-finetune took only three hours.

configuration.yaml:

```
base_model: TinyLlama/TinyLlama-1.1B-step-50K-105b
base_model_config: TinyLlama/TinyLlama-1.1B-step-50K-105b
...
fsdp:
  - full_shard
  - auto_wrap
fsdp_config:
  fsdp_offload_params: true
  fsdp_state_dict_type: FULL_STATE_DICT
  fsdp_transformer_layer_cls_to_wrap: LlamaDecoderLayer
```

Is there any problems here? why does FSDP take 90 hours? I kill the fine-tuning process.
FSDP logs training:
`0%|▎                               | 1/712 [10:58<90:00:48, 447.23s/it]`

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by winglian** [comment]
Using FSDP on tinyllama is a "de-optimization". Tinyllama should already fit onto your gpus so by adding the extra overhead of fsdp, you're simply making it slower. Solution is to remove your fsdp configuration.
